File: server/Logic/Businesses/Business.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

using Logic.Context;
using Logic.Employees;
using Logic.TimeSlots;

namespace Logic.Businesses
{
    public class Business : BaseEntity
    {
        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string Name { get; set; }

        [Required]
        public string OwnerEmail { get; set; }

        [Required]
        public int Zip { get; set; }

        [Required]
        public string Address { get; set; }

        [Required]
        public BusinessType Type { get; set; }

        [Required]
        public int Capacity { get; set; }

        [Required]
        public string Opens { get; set; }

        [Required]
        public string Closes { get; set; }

        [Required]
        public int TimeSlotLength { get; set; }

        public IEnumerable<TimeSlot> TimeSlots { get; set; }

        public IEnumerable<Employee> Employees { get; set; }

    }
}

File: client/src/models/Business.ts
import {DTO} from './General';

export interface BusinessDTO extends DTO {
   address: string;
   name: string;
   zip: string;
   type: string;
   timeSlotLength: number | string;
   capacity: number | string;
   opens: string;
   closes: string;
   ownerEmail?: string;
   longitude?: number;
   latitude?: number;
}

export type BusinessDataDTO = {
   numberOfBookings: number;
   numberOfTimeSlots: number;
   numberOfEmployees?: number;
};

export interface TimeSlotDTO extends DTO {
   id: string;
   date: string;
   start: string;
   end: string;
   business?: string;
}
